metrics/web/test/common/domain/usecases/fetch_instant_config_usecase_test.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:metrics/common/domain/entities/instant_config.dart';
+import 'package:metrics/common/domain/repositories/instant_config_repository.dart';
+import 'package:metrics/common/domain/usecases/fetch_instant_config_usecase.dart';
+import 'package:metrics/common/domain/usecases/parameters/instant_config_param.dart';
+import 'package:mockito/mockito.dart';
+
+// ignore_for_file: avoid_redundant_argument_values
+
+void main() {
+  group("FetchInstantConfigUseCase", () {
+    const isFpsMonitorEnabled = true;
+    const isLoginFormEnabled = false;
+    const isRendererDisplayEnabled = false;
+
+    final repository = _RemoteConfigurationRepositoryMock();
+    final useCase = FetchInstantConfigUseCase(repository);
+    final param = InstantConfigParam(
+      isLoginFormEnabled: isLoginFormEnabled,
+      isFpsMonitorEnabled: isFpsMonitorEnabled,
+      isRendererDisplayEnabled: isRendererDisplayEnabled,
+    );
+
+    tearDown(() {
+      reset(repository);
+    });
+
+    test(
+      "throws an ArgumentError if the given repository is null",
+      () {
+        expect(
+          () => FetchInstantConfigUseCase(null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test("creates an instance with the given repository", () {
+      expect(
+        () => FetchInstantConfigUseCase(repository),
+        returnsNormally,
+      );
+    });
+
+    test(
+      ".call() uses the .fetch() method of the instant config repository",
+      () {
+        useCase(param);
+
+        verify(repository.fetch()).called(1);
+      },
+    );
+
+    test(
+      ".call() returns an instant config with the given param values if the fetched instant config contains null values",
```

Comment:
Should we also add a test case: 
`.call() returns an instant config with the given param values if the fetched instant config is null`